File: LeetCode/9.回文数.py
'''
给你一个整数 x ，如果 x 是一个回文整数，返回 true ；否则，返回 false 。

回文数是指正序（从左向右）和倒序（从右向左）读都是一样的整数。例如，121 是回文，而 123 不是。

来源：力扣（LeetCode）
链接：https://leetcode-cn.com/problems/palindrome-number
著作权归领扣网络所有。商业转载请联系官方授权，非商业转载请注明出处。
'''

class Solution(object):
	def isPalindrome(self, x):
		if x < 0:
			return False
		y = x

		z = 0
		while y != 0:
			z = z * 10 + y % 10
			y = y // 10
		return z == x

File: LeetCode/9.回文数.cpp
class Solution {
public:
    bool isPalindrome(int x) {
		if (x < 0){
			return false;
		}
		int x2 = x;
		int y = 0;
		while (x2 != 0){
			y = y * 10 + x2 % 10;
			x2 = (int)(x2 / 10);
		}
		return x == y;
    }
};